routers/repo/view.go

```diff
@@ -372,6 +372,56 @@ func renderCode(ctx *context.Context) {
 	ctx.Data["TreeLink"] = treeLink
 	ctx.Data["TreeNames"] = treeNames
 	ctx.Data["BranchLink"] = branchLink
+
+	if ctx.Repo.BranchName != ctx.Repo.Repository.DefaultBranch {
+
+		defaultBranch := ctx.Repo.Repository.DefaultBranch
+
+		master, err := ctx.Repo.Repository.GetBranch(defaultBranch)
+		if err != nil {
+			ctx.ServerError("GetBranch", err)
+			return
+		}
+
+		commit, err := master.GetCommit()
+		if err != nil {
+			ctx.ServerError("GetCommit", err)
+			return
+		}
+
+		commitCountInMaster, err := commit.CommitsCount()
+		if err != nil {
+			ctx.ServerError("CommitsCount", err)
+			return
+		}
+
+		n := commitCountInMaster - ctx.Repo.CommitsCount
+
+		var msg string
+
+		if n == 0 {
```

Comment:
how about to use a switch?

Reply:
Or better use functions like in activity page that supports translations based on count